cmd/soroban-rpc/internal/db/transaction.go

```diff
@@ -0,0 +1,387 @@
+package db
+
+import (
+	"context"
+	"database/sql"
+	"encoding/hex"
+	"fmt"
+	"io"
+	"time"
+
+	sq "github.com/Masterminds/squirrel"
+	"github.com/prometheus/client_golang/prometheus"
+
+	"github.com/stellar/go/ingest"
+	"github.com/stellar/go/support/db"
+	"github.com/stellar/go/support/errors"
+	"github.com/stellar/go/support/log"
+	"github.com/stellar/go/xdr"
+
+	"github.com/stellar/soroban-rpc/cmd/soroban-rpc/internal/ledgerbucketwindow"
+)
+
+const (
+	transactionTableName = "transactions"
+)
+
+type Transaction struct {
+	Result           []byte   // XDR encoded xdr.TransactionResult
+	Meta             []byte   // XDR encoded xdr.TransactionMeta
+	Envelope         []byte   // XDR encoded xdr.TransactionEnvelope
+	Events           [][]byte // XDR encoded xdr.DiagnosticEvent
+	FeeBump          bool
+	ApplicationOrder int32
+	Successful       bool
+	Ledger           ledgerbucketwindow.LedgerInfo
+}
+
+// TransactionWriter is used during ingestion
+type TransactionWriter interface {
+	InsertTransactions(lcm xdr.LedgerCloseMeta) error
+	RegisterMetrics(ingest, insert, count prometheus.Observer)
+}
+
+// TransactionReader is used to serve requests and just returns a cloned,
+// read-only DB session to perform actual DB reads on.
+type TransactionReader interface {
+	NewTx(ctx context.Context) (TransactionReaderTx, error)
+	GetLedgerRange(ctx context.Context) ledgerbucketwindow.LedgerRange
```

Comment:
is it necessary to define a GetLedgerRange() on both this interface and in TransactionReaderTx?

Reply:
this was a weird workaround for the `methods.LedgerRangeGetter` interface but I got rid of that entirely in 82ccd3af2dd2bda439d728d41cc71613c85c98cf